### ISSUE TITLE
[NOT AN ISSUE] - Feit Dimmer switches not brightening to 100%

### ISSUE THREAD

**Issue by afraazali** [rospogrigio/localtuya#268]
When using HA to control the lights and bring them up to 100%, the lights don’t go to 100%. Instead I have to go the physical switch and press the button once more to get it to full brightness. Below is my config for a light switch (minus the ids and keys):

```
brightness: 2
brightness_upper: 1000
```

This seems to happen with only some lights. There’s no real pattern. For example, I have 3 sets of track lights, all identical bulbs, but only 1/3 of them don’t work, but they all have the same configuration.

Thank you in advance.

**Comment by ultratoto14** [comment]
@afraazali you can check with DPS dumps if the DP value is well set to max.

As you have multiple devices, are you sure they have the same behavior/values ? (i have two GU10 bulbs from the same manufacturer but i realized after i bought them that one is V1 and the other V2, the first one use a max brightness of 255 where the second is using 1000).

**Comment by lishan89uc** [comment]
Try this also if you they internet access go into the tuya app and see if you can set it to 100% in the app.


```
     - platform: light
        friendly_name: Hallway Lights
        id: 1
        brightness: 2
        brightness_lower: 0
        brightness_upper: 1000
        
```

**Comment by afraazali** [comment]
Here are DPS dumps for the the two lights/dimmers that have this issue:

```
Pot light

On (100% through HA)

Device DEVICE_ID at LOCAL_IP key DEVICE_KEY protocol 3.3 dev_type type_0a:
    DPS [1] VALUE [True]
    DPS [2] VALUE [1000]

On (100% via physical dimmer)

Device DEVICE_ID at LOCAL_IP key DEVICE_KEY protocol 3.3 dev_type type_0a:
    DPS [1] VALUE [True]
    DPS [2] VALUE [1000]

Track Light

On (100% through HA)

Device DEVICE_ID at LOCAL_IP key DEVICE_KEY protocol 3.3 dev_type type_0a:
    DPS [1] VALUE [True]
    DPS [2] VALUE [1000]
    DPS [3] VALUE [10]

On (100% via physical dimmer)

Device DEVICE_ID at LOCAL_IP key DEVICE_KEY protocol 3.3 dev_type type_0a:
    DPS [1] VALUE [True]
    DPS [2] VALUE [1000]
    DPS [3] VALUE [10]
```

It's odd because the dumps are identical whether or not I use HA to control the lights to 100% or use the physical dimmer switch.

**Comment by ultratoto14** [comment]
For the track light, do you have additional feature ? Do you know what is linked to DPS 3 ?

**Comment by afraazali** [comment]
I don't know what's linked to DPS 3. Prior to installing the light switches, they were simply on/off switches. I only found out they were dimmable once I installed the Feit dimmer switches.

**Comment by ultratoto14** [comment]
Sorry but i have no clue, if the value is the same, maybe you swicth have different circuits, using a variable resistor and a relay, meaning that `ON` is different that `DIM 100%` ?

**Comment by lishan89uc** [comment]
> For the track light, do you have additional feature ? Do you know what is linked to DPS 3 ?

If @afraazali has the feit dimmers from costco then dps 3 is the minimum brightness. It is use to prevent dimming the light pass the minimum dimmable amount for some leds.
That said his configuration looks correct. What ha version are you using i remember there was a dimmer fix in one of them in regards to number consistency.

**Comment by afraazali** [comment]
@lishan89uc, I am using the Costco dimmers. Not sure why only this one has the minimum brightness set. In any case, my HA version is 2020.12.1. I’m currently running the Home Assisant OS version on a virtual machine, until I get time to sort out docker on Windows.

**Comment by lishan89uc** [comment]
Oh I think I know what your issue is. This following scenario happen to be before:
I install this dimmer switch on a circuit with 3 florescent lights(6 tubes total). They would flicker even if I set the brightness to 100% in app or on HA. However, if I go to the physical switch and max it, it would stop flashing. I think this might be a hardware/firmware fault. since even in app setting to 100% does nothing.(the lights would still flicker) After that I threw a sonoff on that and use the switch for 2 recessed LED lights made by feit specifically says dimmable. They still flicker at any brightness from 2%-70%. 1% they are fine and from 71% on they would work fine as well. Not sure what else i can do to help. So i am providing you with as much information as i can from my experience with the switches. i have 2 other feit dimmer switches that are working perfect at the moment. they are fitted with feit lightbulbs also from costco also advertised as dimmable.

**Comment by afraazali** [comment]
Thank you @lishan89uc. Is it worth exchanging the switches? I have more of the same switches. I'll try and swap out this switch for another and see if 100% brightness works.

**Comment by lishan89uc** [comment]
@afraazali I bought around 20 switches and they all do the same thing. However 20 switches is still a very small sample size. So you would have to decide that for yourself. I have had a lot of luck changing out the light bulbs though. I can't do anything about the feit recessed lights since they have no bulb but on my changing some of my bulbs to the expensive dimmable leds seem to have done the trick. The ones with no minimal voltage/ a very low one. they can be dimmed to the point where i can only see bulb in the dark room and nothing else... which makes them a bit useless but thats besides the point.

**Comment by afraazali** [comment]
Thank you @lishan89uc. The track lights that don’t work are in the living room, so it might be a good idea to switch them out. This is a new build, so I’m not sure what the quality of the bulbs are. I was going to try switching the non working switch with one that I know works. I’ll report back with my findings.

**Comment by afraazali** [comment]
It’s the bulbs. I switched out the switch for one I know is working properly and that didn’t help. I’m curious to see if the bulbs in the living room track are the same as the kitchen track lights, as those are working fine

**Comment by lishan89uc** [comment]
yes the bulbs are the deciding factor some bulbs need a minimum voltage or need a more stable voltage which this switch can not provide.

**Comment by marcopelegrini** [comment]
@afraazali what API version do you use? I'm having issues figuring out the Feit dimmer configuration, can you post yours?
